orte/mca/oob/tcp/oob_tcp_component.c

```diff
@@ -644,16 +659,37 @@ static void component_shutdown(void)
         /* tell the thread to exit */
         write(mca_oob_tcp_component.stop_thread[1], &i, sizeof(int));
         opal_thread_join(&mca_oob_tcp_component.listen_thread, NULL);
+    } else {
+        opal_output_verbose(2, orte_oob_base_framework.framework_output,
+                        "no hnp or not active");
     }
 
-    while (NULL != (item = opal_list_remove_first(&mca_oob_tcp_component.listeners))) {
-        OBJ_RELEASE(item);
-    }
+    /* because the listeners are in a separate
+     * async thread for apps, we can't just release them here.
+     * Instead, we push it into that event thread and release 
+     * them there */
+     if (ORTE_PROC_IS_APP) {
```

Comment:
@rhc54 let me take it back, the line could be
```
if (ORTE_PROC_IS_APP && orte_initialized)
```

i can reproduce the hang without mpirun (e.g. singleton).
behavior of master and v2.x are differents, and the new test is valid for v2.x only.

if mpirun -np 1 ./a.out
then component_shutdown is invoked at orte_init() time (i really meant init and not finalize)
on both master and v2.x, the async progress is alive and kicking
now, if ./a.out (singleton), then component_shutdown is invoked at orte_finalize() time.
progress_thread is alive on master, but has already finished on v2.x (and hence the hang)

bottom line, replacing
```
if (ORTE_PROC_IS_APP)
```
with
```
if (ORTE_PROC_IS_APP && orte_initialized)
```
works on v2.x only.
The use of ```orte_initialized``` is everything but intuitive here (at least for me) but i could not find a better variable.
i was able to test this on a cray with aprun, and it also removed the hang.

this PR removes an annoying warning i could only observe on OSX.
i can make a new one-off PR for v2.x or wait until the master is stabilized and related commits land into the v2.x branch.

any thoughts ?